### DISCUSSION TITLE
What is the best practice way of thinking about Activities and their scope?

### DISCUSSION THREAD

**Discussion by plolock** [kimai/kimai#5199] [Questions & Answers] [answered]
We are trying to roll out Kimai in our organisation, but are having difficulties deciding how to utilize activities. 

Our first idea was the consider an activity a user story in our accompanying agile platforms (think Jira, Shortcut etc) - but we have to built an integration for that and are not sure if its meant to be a lot activities in the system or not.

The other way around is to just have a few activities, but risking a tad more human error, to timelog most work on fewer activies on global scale - like an activity called "User Story" or "Meeting time" - rather than the specific story or meeting as an activity.

Are there any success stories in adopting Kimai that we can find some inspiration in?

Thank you!

**Comment by kevinpapst** [comment]
Most customer use activities to define the type of work: meeting, development, administration, cleaning, driving, calls, gardening, child care ... highly depends on your industry. 

Where do you want to add rates? 
User, Customer, Project, Activity?

Do you plan to do reporting on stories? 
You could use tags for stories.
But using activities could work as well if your classical structure is:
- customer
- project
- story

The big question is: do you want to achieve anything with them? 
Do you have plans on reporting / invoicing or such on activity level?

**Reply by plolock** [reply]
I see, thank you

What system would you recommend to use in terms of long term use and reporting?
I think what I would want is:

1. Activities are user stories or issue, every logged time/event is the task that was performed related to the story. Multiple per story is possible. Only question here is how scaleable the activities are - as the amount of activities while grow indefinitely.

2.  When exporting the timesheet for a particular customer and project for the previous month, we want to display the name of the activity and all logged tasks and their related time. The current timesheet is good for this, except the activity name is not displayed on it. On the default timesheet it is, which is weird to me.

**Reply by kevinpapst** [reply, marked as answer]
1. I don't know of anyone who uses it like that. But that doesn't mean its wrong. You just have to make sure, to deactivate the activities after a story is closed, so they don't show up in the dropdown (that would dramatically slow down the system after a few months). Give it a try and share your outcome.
2. You are a developer. If you read the docs, you will figure out how to create custom export templates. Or you hire someone who knows the system and who already gives away the software.

**Reply by plolock** [reply]
Excellent, thank you for the support.

We really love Kimai and as we get more familiar with the software, we will probably start recommending it to our customers!

**Reply by hribeir0** [reply]
Sorry to hijack but it's related: is there a way to simply transform Activities in a open text field, not using any model? I would prefer to not have it connected to anything and add them on the fly, as soon as I start tracking time.
Thank you in advance!
